test/coinselector.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { Coin } from '../src/wallet/coin';
+import { CoinSelector } from '../src/wallet/coin-selector.ts';
+import { Transaction } from 'bitcoinjs-lib';
+
+describe('CoinSelector', () => {
+    let coinSelector: CoinSelector;
+    const feeRate: number = 5;
+
+    beforeAll(() => {
+        coinSelector = new CoinSelector(feeRate);
+    });
+
+    it('should generate the correct cost of change', () => {
+        // Fixed value for the cost of change based on the current feeRate.
+        const correctCostOfChange = 495;
+
+        // Assert the correctness of the CoinSelector result.
+        expect(correctCostOfChange).toBe(coinSelector.costOfChange);
+    });
+
+    it.each([
+        {
+            testName: 'should add change to the fee if it is dust',
+            testCoinValues: [
+                1005, 9040, 6440, 2340, 7540, 3920, 5705, 9030, 1092, 5009,
+            ],
+            expectedChange: 0, // Change will be zero when less than dust.
+            expectedCoinCount: 10, // All the coins are going to be selected.
+        },
+        {
+            testName: 'change should be considered when greater than dust',
+            testCoinValues: [5000, 4000],
+            txOutValue: 1000, // Less output so that the change is greater than dust.
+            expectedChange: 2185, // expectedChange = totalSelectedCoin - (txOutValue + transactionFees).
+            expectedCoinCount: 1, // One Coin will suffice for a 1000 sats output for this case.
+        },
```

Comment:
Two things:
1. In the second test case, if UTXO with value 4000, gets selected then the test will fail. It is not getting selected rn because the coin selection is deterministic but once we implement other selection algorithms, this might change. So only use 1 UTXO here or see the amount higher so that both get selected. An interesting case would be when you set `txOutValue` to be 5000.
2. I don't think we need `expectedCoinCount`, we should just be able to use `testCoinValues.length`.

Reply:
Thanks for this edge case. I'll edit the file with only one UTXO.